WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -48,11 +55,17 @@ public void onCreate(Bundle savedInstanceState) {
         } else {
             mSite = (SiteModel) savedInstanceState.getSerializable(WordPress.SITE);
         }
+
+        if (mSite == null) {
+            mSite = mSiteStore.getSiteByLocalId(AppPrefs.getSelectedSite());
```

Comment:
Maybe I missed something, but  `AppPrefs.getSelectedSite` should not be used outside the site picker and the "MySite" fragment.